apiserver/cloneservice/cloneimpl.go

```diff
@@ -0,0 +1,181 @@
+package cloneservice
+
+/*
+Copyright 2019 Crunchy Data Solutions, Inc.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import (
+	"fmt"
+	"io/ioutil"
+	"time"
+
+	crv1 "github.com/crunchydata/postgres-operator/apis/cr/v1"
+	"github.com/crunchydata/postgres-operator/apiserver"
+	msgs "github.com/crunchydata/postgres-operator/apiservermsgs"
+	"github.com/crunchydata/postgres-operator/config"
+	"github.com/crunchydata/postgres-operator/kubeapi"
+	"github.com/crunchydata/postgres-operator/util"
+
+	log "github.com/sirupsen/logrus"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+//  Clone allows a user to clone a cluster into a new deployment
+func Clone(request *msgs.CloneRequest, namespace, pgouser string) msgs.CloneResponse {
+	log.Debugf("clone called with ")
+
+	// set up the response here
+	response := &msgs.CloneResponse{
```

Comment:
@abrightwell The current pattern in the APIs is to return a point to a response-style object, so changing this would represent something going against our current pattern.

That said, I can switch it. I have to pass it in as a pointer to the ultimate JSON function in `cloneservice.go`, but I don't see that as a big deal.

Reply:
Oh ignore my comment...I misread something.